### DISCUSSION TITLE
help with visualizing orientation2color colorkey

### DISCUSSION THREAD

**Discussion by PBurnley** [mtex-toolbox/mtex#2248] [Ask Anything]
Hi there,
I am following the Denoising Orientation Maps tutorial for the purpose of better understanding how the crystal lattice is deforming around grains of a daughter phase growing into a less dense host.  I am able to reproduce the pretty colors in the tutorial that show how the orientation of each pixel varies with respect to the mean orientation of the grain, but I am struggling to figure out what the colors mean.  Ideally I would like to be able to plot a pole figure where the points are colored the same way they are in the image.  I uploaded a little cartoon of my question.  
![Screenshot 2024-10-23 153935](https://github.com/user-attachments/assets/60f14e20-47af-492d-ba17-71a2a334037a)

**Comment by lfgmorales** [comment]
Hi Pamela,

Can you try something like (after loading your ebsd map): and assuming you want to use the ipf colorcode from a map in your pole figures
```matlab
cs = ebsd.CS;
ipfKey = ipfHSVKey(cs);
ipfKey.inversePoleFigureDirection = xvector;

% Step you need -  Assign IPF colors to the orientations
ebsdRGB = ipfKey.orientation2color(ebsd.orientations);
h = Miller(1,0,0,cs);
figure;
plotPDF(ebsd.orientations, ebsdRGB, h, 'MarkerSize',5, 'antipodal');
colorbar;
```
Cheers

Luiz

**Reply by PBurnley** [reply]
Hi Luiz,
This is helpful in that I can produce a pole figure with colored dots! However, the colors are not the same as those in the Denoising Orientation Maps tutorial.  The denoising tutorial produces a suite of colors that are different than the ipf colors (I think its called 'axisAngleColorKey' )  so each grain can have colors across the rainbow even though if you look at the same grain colored according to the ipf color scheme there are just small variations in shade.  There is a description of the axisAngleColorKey but I am struggling a bit with some of the description.  
Cheers
Pamela

**Comment by lfgmorales** [comment]
Hi Pamela

Not sure if this will help and if I remember well, the axisAngleColorKey assigns colors to orientations based on their misorientation axis and angle relative to a reference orientation, using specimen coordinates. The misorientation angle controls the color's saturation: an angle of 0 results in gray, while the maximum angle produces fully saturated colors. The misorientation axis determines the actual color hue. In the X-Y plane, colors cycle through the rainbow, while in the Z direction (±1), they transition to white and black. The reference orientation can either be a fixed single orientation or a list of orientations, such as the average orientation of the grain that the pixel's orientation belongs to. This is more clear if you plot the colorkey itself

that said, you can try to substitute the ipfKey here

ebsdRGB = ipfKey.orientation2color(ebsd.orientations);

by the axisAngleColorkey

But I am not sure if this will work :-D

Cheers

Luiz



Luiz F. G. Morales
ETH Zürich
Structural Geology and Tectonics Group
Scientific Center for Optical and Electron Microscopy (ScopeM)
HPM C 57.3
Otto-Stern-Weg 3
8093 Zurich, Switzerland
Phone +41 44 633 37 46
Email: ***@***.***
